### ISSUE TITLE
More convenience for String Arbritraries

### ISSUE THREAD

**Issue by annam002** [jlink/jqwik#29]
## Testing Problem
I often end up writing code like this.

> Arbitraries.strings().alpha().withChars(new char[]{' ', '"', ',', ';', '.', ':'});

## Suggested Solution

It would be nice if there would be more convenience besides alpha(), ascii() and numeric(), so also something like whitespace() or punctuation(). 
It would also be nice if the withChars() method would take varags, just like the withSamples() method, to get rid of the overhead of creating a char array.

**Comment by jlink** [comment]
`withChars` with varargs is easy. I'll do that.

Could you define which chars you would consider to belong to `whitespace()` and `punctuation()`?

BTW, are you aware that you can easily define your own annotations so that you might not need manual code at all in an example like yours above? Here's the section from the user guide:
http://jqwik.net/user-guide.html#self-made-annotations

**Comment by annam002** [comment]
Cool, looking forward to the varargs functionality. :)
Defining the chars for whitespace could be tricky, since the question is wether to include newlines or not. Other than that I would check which characters are whitespace in Java and use them. For punctuation, I would suggest the usual characters occuring in written language. 
Yes, I am aware that I can write my own annotations, but I think that the suggested methods might be beneficial for other users of jqwik.

**Comment by jlink** [comment]
Implemented `withChars(char... chars)` in https://github.com/jlink/jqwik/commit/4c36f1c6593b6b368d869d84bdf2acc3afaabf03

Change is available from maven ('https://oss.sonatype.org/content/repositories/snapshots') in version 0.8.15-SNAPSHOT

**Comment by jlink** [comment]
Implemented `StringArbitrary.whitespace()` in https://github.com/jlink/jqwik/commit/d96b0a5233ca4cf2c12fb181aa4932c4f58bd086

Change is available from maven ('https://oss.sonatype.org/content/repositories/snapshots') in version 0.8.15-SNAPSHOT

**Comment by jlink** [comment]
Punctuation is a different beast since there is no standard way in Java to determine what a punctuation character is (comparable to `Character.isWhitespace(char)`.

Using all unicode chars allowed according to https://en.wikipedia.org/wiki/General_Punctuation would probably generate characters most people wouldn't expect. Writing your own convenience method like

```
StringArbitrary punctuation(StringArbitrary arbitrary) {
    return arbitrary.withChars('.', ',', ':', ';', '?', '!');
}
```

seems not too much of a hassle.

Therefore, if you don't bring up a strong argument for implementing it despite the general specification problem, I will close this issue as fixed.

**Comment by annam002** [comment]
Ok. Thanks for the quick reaction!

**Comment by jlink** [comment]
Fixed. Will be publicly available in release 0.8.15
